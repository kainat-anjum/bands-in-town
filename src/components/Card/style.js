import styled from "styled-components";
import { Link } from "react-router-dom";
import { device, colors } from "../../commons/variables";

export const CardStyled = styled(Link)`
  display: grid;
  grid-template-columns: auto 1fr;
  grid-gap: 20px;
  align-items: center;
  background: ${colors.darkBlueColor};
  padding: 20px;
  cursor: pointer;

  .artist-image {
    border-radius: 50%;
    height: 100px;
    width: 100px;
    object-fit: cover;
  }

  .artist-name {
    color: ${colors.whiteColor};
    font-weight: 500;
    margin-bottom: 0;
  }

  @media ${device.mobileL} {
    padding: 12px;
  }
`;
