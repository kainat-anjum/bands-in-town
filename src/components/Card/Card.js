import { React } from "react";
import { CardStyled } from "./style";

export const Card = (props) => {
  const { image, name } = props;

  return (
    <CardStyled to="/artist">
      <img className="artist-image" src={image} alt="artist-image" />
      <p className="artist-name">{name}</p>
    </CardStyled>
  );
};
