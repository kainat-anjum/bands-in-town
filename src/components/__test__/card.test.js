import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Card } from "../Card/Card";

const data = {
  name: "Mario",
  image: "https://photos.bandsintown.com/large/11058086.jpeg",
};

afterEach(cleanup);

describe("Testing Card Data", () => {
  it("Tests name and image", () => {
    const { image, name } = data;
    const tree = renderer
      .create(
        <Router>
          <Card image={image} name={name} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
