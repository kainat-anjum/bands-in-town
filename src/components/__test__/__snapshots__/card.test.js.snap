// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Data Tests name and image 1`] = `
<a
  className="sc-bdvvaa iNgzZQ"
  href="/artist"
  onClick={[Function]}
>
  <img
    alt=""
    className="artist-image"
    src="https://photos.bandsintown.com/large/11058086.jpeg"
  />
  <p
    className="artist-name"
  >
    Mario
  </p>
</a>
`;
